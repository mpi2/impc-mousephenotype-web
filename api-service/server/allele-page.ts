import { fetchAPIFromServer, fetchURL } from "@/api-service";
import { AlleleSummary } from "@/models";
const KUBERNETES_NAMESPACE = process.env.KUBERNETES_NAMESPACE ?? "default";
const NODE_ENV = process.env.NODE_ENV;

export async function fetchAlleleSummary(
  mgiGeneAccessionId: string,
  alleleName: string,
): Promise<AlleleSummary> {
  const endpointURL = `http://impc-allele-service.${KUBERNETES_NAMESPACE}:8080/v1/alleles?mgiGeneAccessionId=${mgiGeneAccessionId}&alleleName=${alleleName}`;
  return await (NODE_ENV === "development"
    ? fetchAPIFromServer<AlleleSummary>(
        `/api/v1/alleles/${mgiGeneAccessionId}/${alleleName}`,
      )
    : fetchURL<AlleleSummary>(endpointURL));
}
