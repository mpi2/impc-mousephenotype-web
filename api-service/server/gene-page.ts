import { GeneOrder, GenePhenotypeHits, GeneSummary } from "@/models/gene";
import { fetchAPIFromServer, fetchURL } from "@/api-service";
import { processGeneOrderResponse } from "@/hooks/gene-order.query";
import { processGenePhenotypeHitsResponse } from "@/hooks/significant-phenotypes.query";

const KUBERNETES_NAMESPACE = process.env.KUBERNETES_NAMESPACE ?? "default";
const NODE_ENV = process.env.NODE_ENV;

export async function fetchGeneSummary(
  mgiGeneAccessionId: string,
): Promise<GeneSummary> {
  const endpointURL = `http://impc-summary-service.${KUBERNETES_NAMESPACE}:8080/v1/gene_summary?mgiGeneAccessionId=${mgiGeneAccessionId}`;
  return await (NODE_ENV === "development"
    ? fetchAPIFromServer<GeneSummary>(
        `/api/v1/genes/${mgiGeneAccessionId}/summary`,
      )
    : fetchURL<GeneSummary>(endpointURL));
}

export async function fetchGenePhenotypeHits(
  mgiGeneAccessionId: string,
): Promise<Array<GenePhenotypeHits>> {
  const endpointURL = `http://impc-phenotype-hits-service.${KUBERNETES_NAMESPACE}:8080/v1/phenotypehits?mgiGeneAccessionId=${mgiGeneAccessionId}`;
  const data = await (NODE_ENV === "development"
    ? fetchAPIFromServer<Array<GenePhenotypeHits>>(
        `/api/v1/genes/${mgiGeneAccessionId}/phenotype-hits`,
      )
    : fetchURL<Array<GenePhenotypeHits>>(endpointURL));
  return processGenePhenotypeHitsResponse(data);
}

export async function fetchGeneOrderData(
  mgiGeneAccessionId: string,
): Promise<Array<GeneOrder>> {
  const endpointURL = `http://impc-order-service.${KUBERNETES_NAMESPACE}:8080/v1/orders?mgiGeneAccessionId=${mgiGeneAccessionId}`;
  const data = await (NODE_ENV === "development"
    ? fetchAPIFromServer<Array<GeneOrder>>(
        `/api/v1/genes/${mgiGeneAccessionId}/order`,
      )
    : fetchURL<Array<GeneOrder>>(endpointURL));
  return processGeneOrderResponse(data);
}
