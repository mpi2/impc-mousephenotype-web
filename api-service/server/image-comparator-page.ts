import { GeneImageCollection } from "@/models/gene";
import { fetchAPIFromServer, fetchURL } from "@/api-service";

const KUBERNETES_NAMESPACE = process.env.KUBERNETES_NAMESPACE ?? "default";
const NODE_ENV = process.env.NODE_ENV;

export async function fetchMutantImages(
  mgiGeneAccessionId: string,
  parameterStableId: string,
): Promise<Array<GeneImageCollection>> {
  const endpointURL = `http://impc-imaging-service.${KUBERNETES_NAMESPACE}:8080/v1/images/findByMgiAndStableId?mgiGeneAccessionId=${mgiGeneAccessionId}&parameterStableId=${parameterStableId}`;
  try {
    return await (NODE_ENV === "development"
      ? fetchAPIFromServer<Array<GeneImageCollection>>(
          `/api/v1/images/find_by_mgi_and_stable_id?mgiGeneAccessionId=${mgiGeneAccessionId}&parameterStableId=${parameterStableId}`,
        )
      : fetchURL<Array<GeneImageCollection>>(endpointURL));
  } catch (error) {
    return [];
  }
}

export async function fetchControlImages(
  parameterStableId: string,
): Promise<Array<GeneImageCollection>> {
  const endpointURL = `http://impc-imaging-service.${KUBERNETES_NAMESPACE}:8080/v1/images/findByStableIdAndSampleId?biologicalSampleGroup=control&parameterStableId=${parameterStableId}`;
  try {
    return await (NODE_ENV === "development"
      ? fetchAPIFromServer<Array<GeneImageCollection>>(
          `/api/v1/images/find_by_stable_id_and_sample_id?biologicalSampleGroup=control&parameterStableId=${parameterStableId}`,
        )
      : fetchURL<Array<GeneImageCollection>>(endpointURL));
  } catch (error) {
    return [];
  }
}
