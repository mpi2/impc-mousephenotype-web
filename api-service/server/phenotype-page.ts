import { fetchAPIFromServer, fetchURL } from "@/api-service";
import { PhenotypeSummary } from "@/models/phenotype";

const KUBERNETES_NAMESPACE = process.env.KUBERNETES_NAMESPACE ?? "default";
const NODE_ENV = process.env.NODE_ENV;

export async function fetchPhenotypeSummary(
  phenotypeId: string,
): Promise<PhenotypeSummary> {
  const endpointURL = `http://impc-summary-service.${KUBERNETES_NAMESPACE}:8080/v1/phenotype_summary?phenotypeId=${phenotypeId}`;
  return await (NODE_ENV === "development"
    ? fetchAPIFromServer<PhenotypeSummary>(
        `/api/v1/phenotypes/${phenotypeId}/summary`,
      )
    : fetchURL<PhenotypeSummary>(endpointURL));
}
