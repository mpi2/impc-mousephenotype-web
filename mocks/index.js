export async function initMocks() {
  if (typeof window === "undefined") {
    const { server } = await import("./server");
    server.listen();
  } else {
    const { worker } = await import("./browser");
    return worker.start({
      serviceWorker: {
        url: "/data/mockServiceWorker.js",
        options: {
          scope: "/",
        }
      }
    });
  }
}

