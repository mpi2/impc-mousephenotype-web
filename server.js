const { createServer } = require('http');
const { parse } = require('url');
const next = require('next');

const dev = process.env.NODE_ENV !== 'production';
const hostname = 'localhost';
const port = 3000;
const app = next({ dev, hostname, port });
const handle = app.getRequestHandler();

app.prepare()
.then(() => {
    createServer(async (req, res) => {
        try {
            const parsedUrl = parse(req.url, true);
            const { pathname, query} = parsedUrl;
            const host = req.headers.host;
            console.log({ host, pathname });
            res.setHeader('X-IMPC-TEST', "true");
            if (host.includes('nginx.mousephenotype-dev.org')) {
                const modifiedPathName = pathname.replace("/data/", "");
                console.log({ modifiedPathName });
                await app.render(req, res, modifiedPathName, query);
                res.setHeader('X-PATH', JSON.stringify({ host, pathname, modifiedPathName }));
            } else {
                await handle(req, res, parsedUrl);
            }

        } catch (err) {
            console.error('Error occurred handling', req.url, err)
            res.statusCode = 500
            res.end('internal server error')
        }
    })
        .once('error', (err) => {
            console.error(err);
            process.exit(1);
        })
        .listen(port, () => {
            console.log(`Ready on http://${hostname}:${port}`);
        })
});