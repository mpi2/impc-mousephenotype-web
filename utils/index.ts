import _ from "lodash";

export const formatBodySystems = (systems: string[] | string = []) => {
  return _.capitalize(
    (typeof systems === "string" ? systems : systems.join(", "))
      .replace(/ phenotype/g, "")
      .replace(/\//g, " / ")
  );
};

export const formatAlleleSymbol = (allele: string) => {
  return allele.slice(0, allele.length - 1).split("<");
};
