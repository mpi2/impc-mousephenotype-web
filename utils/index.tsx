import React from "react";
import _ from "lodash";

export const formatBodySystems = (systems: string[] | string = []) => {
  return _.capitalize(
    (typeof systems === "string" ? systems : systems.join(", "))
      .replace(/ phenotype/g, "")
      .replace(/\//g, " / ")
  );
};

export const formatAlleleSymbol = (allele: string) => {
  return allele.slice(0, allele.length - 1).split("<");
};

export const formatPValue = (pValue: number) => {
  const pValueArray = Number.parseFloat(String(pValue))
    .toExponential(2)
    .split("e");
  return (
    <>
      {pValueArray[0]}x10<sup>{pValueArray[1].replace("+", "")}</sup>
    </>
  );
};

export const formatESCellName = (src: string) => {
  const [name, superscript, end] = src.split(/<|>/);
  return (
    <>
      {name}
      <sup>{superscript}</sup>
      {end}
    </>
  );
};

const fetchCache = {};
export { fetchCache };

export const toSentenceCase = (camelCase) => {
  if (camelCase) {
    const result = camelCase.replace(/([A-Z])/g, " $1");
    return result[0].toUpperCase() + result.substring(1).toLowerCase();
  }
  return "";
};
