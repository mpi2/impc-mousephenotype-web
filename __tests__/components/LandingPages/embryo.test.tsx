import renderer from 'react-test-renderer';
import EmbryoLandingPage from "@/pages/embryo";
import { server } from "../../../mocks/server";
import { rest } from "msw";
import { API_URL, createTestQueryClient } from "../../utils";
import { QueryClientProvider } from "@tanstack/react-query";

jest.mock('next/router', () => jest.requireActual('next-router-mock'));

describe('Embryo landing page', () => {
  it('renders correctly', () => {
    const client = createTestQueryClient();
    server.use(
      rest.get(`${API_URL}/api/v1/publications`, (req, res, ctx) => {
        return res(ctx.status(200), ctx.json({
          content: [],
          first: true,
          last: false,
          number: 0,
          numberOfElements: 0,
          size: 10,
          totalElements: 0,
          totalPages: 0,
        }));
      })
    );
    const tree = renderer.create(
      <QueryClientProvider client={client}>
        <EmbryoLandingPage />
      </QueryClientProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});