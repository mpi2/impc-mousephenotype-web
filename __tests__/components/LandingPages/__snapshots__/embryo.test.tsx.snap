// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Embryo landing page renders correctly 1`] = `
[
  <div
    className="banner"
  >
    <div
      className="pb-4 pt-5 container container"
    >
      <div
        className="col-12 col-md-8 ps-4 pe-4"
      >
        <div
          className="tabs"
        >
          <a
            className="tab"
            href="/search"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            Genes
          </a>
          <a
            className="tab"
            href="/search?type=phenotype"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            Phenotypes
          </a>
          <a
            className="tab"
            href="https://dev.mousephenotype.org/?s="
          >
            Help, news, blog
          </a>
        </div>
        <div
          className="inputCont"
        >
          <input
            className="input"
            onChange={[Function]}
            onKeyDown={[Function]}
            placeholder="Search All 7824 Knockout Data..."
            type="text"
            value=""
          />
          <button
            className="searchBtn"
            onClick={[Function]}
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-magnifying-glass "
              data-icon="magnifying-glass"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={{}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z"
                fill="currentColor"
                style={{}}
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="page container"
    style={
      {
        "lineHeight": 2,
      }
    }
  >
    <div
      className="card"
    >
      <div
        className="subheading"
      >
        <nav
          aria-label="breadcrumb"
        >
          <ol
            className="breadcrumb"
          >
            <li
              aria-current="page"
              className="breadcrumb-item active"
            >
              Home
            </li>
            <li
              aria-current="page"
              className="breadcrumb-item active"
            >
              IMPC data collections
            </li>
            <li
              aria-current="page"
              className="breadcrumb-item active"
            >
              Embryo Data
            </li>
          </ol>
        </nav>
      </div>
      <h1
        className="mb-4 mt-2"
      >
        <strong>
          Introduction to IMPC Embryo Data
        </strong>
      </h1>
      <div
        className="container"
      >
        <div
          className="row"
        >
          <div
            className="text-center col-md-4 col-12"
          >
            <img
              className="embryoImage img-fluid"
              src="images/landing-pages/Tmem100_het.jpg"
            />
          </div>
          <div
            className="col-md-8 col-12"
          >
            <p>
              Up to one third of homozygous knockout lines are lethal, which means no homozygous mice or less than expected are observed past the weaning stage (IMPC 
              <a
                className="link primary"
                href="https://beta.mousephenotype.org/impress/ProcedureInfo?action=list&procID=703&pipeID=7"
              >
                Viability Primary Screen procedure
              </a>
              ). Early death may occur during embryonic development or soon after birth, during the pre-weaning stage. For this reason, the IMPC established a 
              <a
                className="link primary"
                href="https://beta.mousephenotype.org/impress"
              >
                systematic embryonic phenotyping pipeline
              </a>
               to morphologically evaluate mutant embryos to ascertain the primary perturbations that cause early death and thus gain insight into gene function.
            </p>
          </div>
        </div>
        <div
          className="mt-4 row"
        >
          <div
            className="col"
          >
            <p>
              Read more in our paper on 
              <a
                className="link primary"
                href="https://europepmc.org/article/PMC/5295821"
              >
                High-throughput discovery of novel developmental phenotypes, Nature 2016.
              </a>
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      className="card"
    >
      <div
        className="container"
      >
        <h1>
          <strong>
            IMPC Embryonic Pipeline
          </strong>
        </h1>
        <p>
          As determined in IMPReSS (see interactive diagram here), all embryonic lethal lines undergo gross morphology assessment at E12.5 (embryonic day 12.5) to determine whether defects occur earlier or later during embryonic development. A comprehensive imaging platform is then used to assess dysmorphology. Embryo gross morphology, as well as 2D and 3D imaging are actively being implemented by the IMPC for lethal lines.
        </p>
        <div
          className="text-center"
        >
          <img
            className="img-fluid"
            src="images/landing-pages/IMPC-Embryo-Pipeline-Diagram.png"
          />
        </div>
      </div>
    </div>
    <div
      className="card"
    >
      <div
        className="container"
      >
        <h1>
          <strong>
            Determining Lethal Lines
          </strong>
        </h1>
        <p>
          The IMPC assesses each gene knockout line for viability (Viability Primary Screen  
          <a
            className="link primary"
            href="https://beta.mousephenotype.org/impress/ProcedureInfo?action=list&procID=703&pipeID=7"
          >
            IMPC_VIA_001
          </a>
           and 
          <a
            className="link primary"
            href="https://beta.mousephenotype.org/impress/ProcedureInfo?action=list&procID=1188&pipeID=7#Parameters"
          >
            IMPC_VIA_002
          </a>
          ). In this procedure, the proportion of homozygous pups is determined soon after birth, during the preweaning stage, in litters produced from mating heterozygous animals. A line is declared lethal if no homozygous pups for the null allele are detected at weaning age, and subviable if pups homozygous for the null allele constitute less than 12.5% of the litter.
        </p>
        <div
          className="mb-3 row"
        >
          <div
            className="col-md-7"
          >
            <div
              className="chartWrapper"
            >
              <canvas
                height={150}
                role="img"
                width={300}
              />
            </div>
          </div>
          <div
            className="col-md-5"
          >
            <div
              className="tableWrapper"
            >
              <table
                className="table striped table table-bordered"
              >
                <thead>
                  <tr>
                    <th
                      rowSpan={2}
                      width="8.333333333333332%"
                    >
                      <button
                        className="inlineButton headerButton"
                        onClick={[Function]}
                        style={
                          {
                            "fontWeight": "normal",
                          }
                        }
                      >
                        Category
                         
                      </button>
                    </th>
                    <th
                      rowSpan={2}
                      width="8.333333333333332%"
                    >
                      <button
                        className="inlineButton headerButton"
                        onClick={[Function]}
                        style={
                          {
                            "fontWeight": "normal",
                          }
                        }
                      >
                        Lines
                         
                      </button>
                    </th>
                  </tr>
                </thead>
                <tbody>
                  <tr>
                    <td>
                      Viable
                    </td>
                    <td>
                      4897
                    </td>
                  </tr>
                  <tr>
                    <td>
                      Lethal
                    </td>
                    <td>
                      1859
                    </td>
                  </tr>
                  <tr>
                    <td>
                      Subviable
                    </td>
                    <td>
                      886
                    </td>
                  </tr>
                  <tr>
                    <td
                      colSpan={2}
                    >
                      <a
                        className="link primary"
                        href="https://ftp.ebi.ac.uk/pub/databases/impc/all-data-releases/latest/results/viability.csv.gz"
                      >
                        Download
                      </a>
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
        <div
          className="row"
        >
          <p>
            Lethal strains are further phenotyped in the 
            <a
              className="link primary"
              href="https://beta.mousephenotype.org/impress"
            >
              embryonic phenotyping pipeline
            </a>
            . For embryonic lethal and subviable strains, heterozygotes are phenotyped in the IMPC 
            <a
              className="link primary"
              href="https://beta.mousephenotype.org/impress"
            >
              adult phenotyping pipeline
            </a>
            .
          </p>
        </div>
        <div
          className="row"
        >
          <div
            className="col-md-7"
          >
            <div
              className="chartWrapper"
            >
              <canvas
                height={150}
                role="img"
                width={300}
              />
            </div>
          </div>
          <div
            className="col-md-5"
          >
            <div
              className="tableWrapper"
            >
              <table
                className="table striped table table-bordered"
              >
                <thead>
                  <tr>
                    <th
                      rowSpan={2}
                      width="8.333333333333332%"
                    >
                      <button
                        className="inlineButton headerButton"
                        onClick={[Function]}
                        style={
                          {
                            "fontWeight": "normal",
                          }
                        }
                      >
                        Category
                         
                      </button>
                    </th>
                    <th
                      rowSpan={2}
                      width="8.333333333333332%"
                    >
                      <button
                        className="inlineButton headerButton"
                        onClick={[Function]}
                        style={
                          {
                            "fontWeight": "normal",
                          }
                        }
                      >
                        Lines
                         
                      </button>
                    </th>
                  </tr>
                </thead>
                <tbody>
                  <tr>
                    <td>
                      Perinatal lethal
                    </td>
                    <td>
                      326
                    </td>
                  </tr>
                  <tr>
                    <td>
                      E9.5 lethal
                    </td>
                    <td>
                      450
                    </td>
                  </tr>
                  <tr>
                    <td>
                      E12.5 lethal
                    </td>
                    <td>
                      97
                    </td>
                  </tr>
                  <tr>
                    <td>
                      E15.5 lethal
                    </td>
                    <td>
                      20
                    </td>
                  </tr>
                  <tr>
                    <td>
                      E18.5 lethal
                    </td>
                    <td>
                      6
                    </td>
                  </tr>
                  <tr>
                    <td>
                      Lorem ipsum
                    </td>
                    <td>
                      143
                    </td>
                  </tr>
                  <tr>
                    <td
                      colSpan={2}
                    >
                      <a
                        className="link primary"
                        href="https://impc-datasets.s3.eu-west-2.amazonaws.com/embryo-landing-assets/wol_all.csv"
                      >
                        Download
                      </a>
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="card"
    >
      <div
        className="container"
      >
        <h1>
          <strong>
            Embryo Data Availability Grid
          </strong>
        </h1>
        <div
          className="row"
        >
          <div
            className="col"
          >
            <p>
              Filter by Window of Lethality
            </p>
            <div
              className="row m-2 "
            >
              <div
                className="col-6"
              >
                <div
                  className="basic-multi-select css-b62m3t-container"
                  onKeyDown={[Function]}
                >
                  <span
                    className="css-1f43avz-a11yText-A11yText"
                    id="react-select-2-live-region"
                  />
                  <span
                    aria-atomic="false"
                    aria-live="polite"
                    aria-relevant="additions text"
                    className="css-1f43avz-a11yText-A11yText"
                    role="log"
                  />
                  <div
                    className="select__control css-13cymwt-control"
                    onMouseDown={[Function]}
                    onTouchEnd={[Function]}
                  >
                    <div
                      className="select__value-container select__value-container--is-multi css-1fdsijx-ValueContainer"
                    >
                      <div
                        className="select__placeholder css-1jqq78o-placeholder"
                        id="react-select-2-placeholder"
                      >
                        Select window of lethality
                      </div>
                      <div
                        className="select__input-container css-qbdosj-Input"
                        data-value=""
                      >
                        <input
                          aria-activedescendant=""
                          aria-autocomplete="list"
                          aria-describedby="react-select-2-placeholder"
                          aria-expanded={false}
                          aria-haspopup={true}
                          autoCapitalize="none"
                          autoComplete="off"
                          autoCorrect="off"
                          className="select__input"
                          disabled={false}
                          id="react-select-2-input"
                          onBlur={[Function]}
                          onChange={[Function]}
                          onFocus={[Function]}
                          role="combobox"
                          spellCheck="false"
                          style={
                            {
                              "background": 0,
                              "border": 0,
                              "color": "inherit",
                              "font": "inherit",
                              "gridArea": "1 / 2",
                              "label": "input",
                              "margin": 0,
                              "minWidth": "2px",
                              "opacity": 1,
                              "outline": 0,
                              "padding": 0,
                              "width": "100%",
                            }
                          }
                          tabIndex={0}
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                    <div
                      className="select__indicators css-1hb7zxy-IndicatorsContainer"
                    >
                      <span
                        className="select__indicator-separator css-1u9des2-indicatorSeparator"
                      />
                      <div
                        aria-hidden="true"
                        className="select__indicator select__dropdown-indicator css-1xc3v61-indicatorContainer"
                        onMouseDown={[Function]}
                        onTouchEnd={[Function]}
                      >
                        <svg
                          aria-hidden="true"
                          className="css-tj5bde-Svg"
                          focusable="false"
                          height={20}
                          viewBox="0 0 20 20"
                          width={20}
                        >
                          <path
                            d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="col-6"
              />
            </div>
            <div
              style={
                {
                  "backgroundColor": "white",
                  "height": "600px",
                  "marginLeft": "0",
                  "marginRight": "0",
                  "marginTop": "0",
                }
              }
            >
              <div
                style={
                  {
                    "height": "100%",
                    "width": "100%",
                  }
                }
              />
            </div>
            <nav
              aria-label="Page navigation example"
              style={{}}
            >
              <ul
                className="pagination justify-content-center"
              >
                <li
                  className="page-item disabled"
                >
                  <button
                    aria-label="Previous"
                    className="page-link"
                    onClick={[Function]}
                  >
                    <span
                      aria-hidden="true"
                    >
                      «
                    </span>
                  </button>
                </li>
                <li
                  className="page-item active"
                >
                  <button
                    className="page-link"
                    onClick={[Function]}
                  >
                    1
                  </button>
                </li>
                <li
                  className="page-item "
                >
                  <button
                    className="page-link"
                    onClick={[Function]}
                  >
                    2
                  </button>
                </li>
                <li
                  className="page-item "
                >
                  <button
                    className="page-link"
                    onClick={[Function]}
                  >
                    3
                  </button>
                </li>
                <li
                  className="page-item disabled"
                >
                  <span
                    className="page-link"
                  >
                    ...
                  </span>
                </li>
                <li
                  className="page-item "
                >
                  <button
                    aria-label="Previous"
                    className="page-link"
                    onClick={[Function]}
                  >
                    <span
                      aria-hidden="true"
                    >
                      32
                    </span>
                  </button>
                </li>
                <li
                  className="page-item "
                >
                  <button
                    aria-label="Next"
                    className="page-link"
                    onClick={[Function]}
                  >
                    <span
                      aria-hidden="true"
                    >
                      »
                    </span>
                  </button>
                </li>
              </ul>
            </nav>
          </div>
        </div>
      </div>
    </div>
    <div
      className="card"
    >
      <div
        className="container"
      >
        <h1>
          <strong>
            Accessing Embryo Phenotype Data
          </strong>
        </h1>
        <div
          className="row"
        >
          <div
            className="col"
          >
            <p>
              Embryo phenotye data can be accessed in multiple ways:
            </p>
            <ul>
              <li>
                <a
                  className="link primary"
                  href="https://beta.mousephenotype.org/data/embryo_imaging"
                >
                  Embryo Images: interactive heatmap
                </a>
                 A compilation of all our Embryo Images, organised by gene and life stage, with access to the Interactive Embryo Viewer, where you can compare mutants and wild types side by side and rotate 2D and 3D images; we also provide access to our external partners' embryo images.
              </li>
              <li>
                <a
                  className="link primary"
                  href="https://beta.mousephenotype.org/data/embryo/vignettes"
                >
                  Embryo Vignettes
                </a>
                 Showcase of best embryo images with detailed explanations.
              </li>
              <li>
                <a
                  className="link primary"
                  href="ftp://ftp.ebi.ac.uk/pub/databases/impc/all-data-releases/latest/results/"
                >
                  From the FTP site, latest release
                </a>
                 All our results. Reports need to be filtered by a dedicated column, Life Stage (E9.5, E12.5, E15.5 and E18.5). Please check the README file or see documentation here.
              </li>
              <li>
                Using the REST API (see documentation 
                <a
                  className="link primary"
                  href="https://www.mousephenotype.org/help/programmatic-data-access/"
                >
                  here
                </a>
                )
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
    <div
      className="card"
    >
      <div
        className="container"
      >
        <h1>
          <strong>
            IKMC/IMPC related publications
          </strong>
        </h1>
      </div>
    </div>
  </div>,
]
`;
